src/io/platform/f3xx/f302x8/CANf302x8.cpp

```diff
@@ -101,19 +101,31 @@ CANf302x8::CANf302x8(Pin txPin, Pin rxPin, bool loopbackEnabled)
     HAL_NVIC_SetPriority(CAN_RX0_IRQn, EVT::core::platform::CAN_INTERRUPT_PRIORITY, 0);
     HAL_NVIC_EnableIRQ(CAN_RX0_IRQn);
 
-    /* By default - filter that accepts all incoming messages */
-    CAN_FilterTypeDef defaultFilter;
-    defaultFilter.FilterIdHigh = 0;
-    defaultFilter.FilterIdLow = 0;
-    defaultFilter.FilterMaskIdHigh = 0;
-    defaultFilter.FilterMaskIdLow = 0;
-    defaultFilter.FilterFIFOAssignment = CAN_FILTER_FIFO0;
-    defaultFilter.FilterBank = 0;
-    defaultFilter.FilterMode = CAN_FILTERMODE_IDMASK;
-    defaultFilter.FilterScale = CAN_FILTERSCALE_32BIT;
-    defaultFilter.FilterActivation = ENABLE;
-
-    HAL_CAN_ConfigFilter(&halCAN, &defaultFilter);
+    CAN_FilterTypeDef emergencyFilter;
```

Comment:
For these filters, the user of the code should be able to pass these in more explicitly. Filtering based on ID is good for the case of getting the messages targeting a specific device, but we also want the ability to letting in messages coming from another device. Ultimately what we would want is something along the lines of

```
can.addExplicitFilter(0x05); // Will let in messages with the CAN ID 0x05
can.addExplicitFilter(0x06); // Will let in message with the CAN ID 0x06
can.addMaskFilter(0x83); // Will add this mask as a filter
```

That way we can say explicit allow for the TMS to get messages that correspond to BMS temperature data. Additionally this will apply better in places were we use CAN, but not necessarily CANopen (like working with the ground fault detection board).